packages/expo-video/ios/VideoView.swift

```diff
@@ -29,6 +29,18 @@ public final class VideoView: ExpoView, AVPlayerViewControllerDelegate {
   }
   #endif
 
+  #if os(tvOS)
+  let allowsVideoFrameAnalsysis = false
+  #else
+  var allowsVideoFrameAnalsysis: Bool = true {
+    didSet {
+      if #available(iOS 16.0, macCatalyst 18.0, *) {
+        playerViewController.allowsVideoFrameAnalysis = allowsVideoFrameAnalsysis
+      }
+    }
```

Comment:
We found an ugly workaround and couldn't come up with anything better. At least it seems to work well.

```suggestion
  var allowsVideoFrameAnalsysis: Bool = true {
    didSet {
      if #available(iOS 16.0, macCatalyst 18.0, *) {
        forceAllowsVideoFrameAnalysis(allowsVideoFrameAnalsysis)
      }
    }
```
And add a function under `func stopPictureInPicture()..` 

```swift
  @available(iOS 16.0, *)
  private func forceAllowsVideoFrameAnalysis(_ allowsVideoFrameAnalsysis: Bool) {
    playerViewController.allowsVideoFrameAnalysis = allowsVideoFrameAnalsysis

    // Setting the `allowsVideoFrameAnalysis` to false doesn't update the UI. We can force the desired behaviour by quickly toggling the property.
    // Setting it to true clears existing requests, which updates the UI, hiding the button, then setting it to false before it detects any text keeps it
    // in the desired state.
    // Tested in iOS 17.5
    if (!allowsVideoFrameAnalsysis) {
      playerViewController.allowsVideoFrameAnalysis = true
      playerViewController.allowsVideoFrameAnalysis = false
    }
  }
  ```

Reply:
That works great! Nice find @behenate